test/coverage/datamodel/UUriSerializerTest.cpp

```diff
@@ -23,16 +23,167 @@ class TestFixture : public testing::Test {
+	const std::string expectedUUri = "//192.168.1.10/1100/FF/8000";
+	const std::string actualUUri =
+	    uprotocol::datamodel::serializer::uri::AsString::serialize(*testUUri);
+	ASSERT_EQ(expectedUUri, actualUUri);
+}
+
+// Positive test case - test serialization of UUri with no authority to string
+TEST_F(TestUUriSerializaerFixture, SerializeUUriWithNoAuthorityToString) {
+	const std::shared_ptr<uprotocol::v1::UUri> testUUri = buildValidTestURI("");
+	const std::string expectedUUri = "/1100/FF/8000";
+	const std::string actualUUri =
+	    uprotocol::datamodel::serializer::uri::AsString::serialize(*testUUri);
+	ASSERT_EQ(expectedUUri, actualUUri);
+}
+
+// TODO: Enable this test once validator for UUri is implemented.
+// TEST_F(TestUUriSerializaerFixture, SerializeEmptyUUriToString) {
+// 	uprotocol::v1::UUri testUUri ;
+// 	std::string expectedUUri = "";
+// 	std::string actualUUri 	 =
+// uprotocol::datamodel::serializer::uri::AsString::serialize(testUUri);
+// 	ASSERT_EQ(expectedUUri, actualUUri);
+// }
+
+TEST_F(TestUUriSerializaerFixture, DeSerializeUUriStringWithScheme) {
+	const std::string uuriAsString = "up://192.168.1.10/1100/FF/8000";
+	const std::string expectedAuthority = "192.168.1.10";
+	const uint32_t expectedUEID = 0x1100;
+	const uint32_t expectedMajorVersion = 0xFF;
+	const uint32_t expectedResourceID = 0x8000;
+
+	const uprotocol::v1::UUri uri =
+	    uprotocol::datamodel::serializer::uri::AsString::deserialize(
+	        uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
+
+TEST_F(TestUUriSerializaerFixture, DeSerializeUUriStringWithoutScheme) {
+	const std::string uuriAsString = "//192.168.1.10/1100/FF/8000";
+	const std::string expectedAuthority = "192.168.1.10";
+	const uint32_t expectedUEID = 0x1100;
+	const uint32_t expectedMajorVersion = 0xFF;
+	const uint32_t expectedResourceID = 0x8000;
+
+	const uprotocol::v1::UUri uri =
+	    uprotocol::datamodel::serializer::uri::AsString::deserialize(
+	        uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
+
+TEST_F(TestUUriSerializaerFixture, DeSerializeEmptyUUriString) {
+	const std::string uuriAsString = "";
+	const std::string expectedAuthority = "";
+	const uint32_t expectedUEID = 0x0;
+	const uint32_t expectedMajorVersion = 0x0;
+	const uint32_t expectedResourceID = 0x0;
+
+	const uprotocol::v1::UUri uri =
+	    uprotocol::datamodel::serializer::uri::AsString::deserialize(
+	        uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
```

Comment:
Shouldn't this throw an exception since its not a valid URI?

Reply:
I thought the same initially but then when I referred java implementation, there was no exception thrown and instead just empty string was returned. I can add exception handling here . Let me know what you suggest.

Reply:
Thanks for explanation. I agree with 1st approach which is throwing an exception for this use-case. I have added exception throw for empty string and also updated testcase.